src/aggregator/aggregator/handler/writer/protobuf.go

```diff
@@ -198,3 +200,69 @@ func (f storagePolicyFilter) Filter(m producer.Message) bool {
 	}
 	return false
 }
+
+// RoutingPolicyFilterParams provides parameters for creating a routing policy filter.
+type RoutingPolicyFilterParams struct {
+	Logger               *zap.Logger
+	RoutingPolicyHandler routing.PolicyHandler
+	IsDefault            bool
+	AllowedTrafficTypes  []string
+}
+
+// NewRoutingPolicyFilter creates a new routing policy based filter.
+func NewRoutingPolicyFilter(
+	p RoutingPolicyFilterParams, configSource producer.FilterFuncConfigSourceType,
+) producer.FilterFunc {
+	logger := p.Logger
+	if logger == nil {
+		logger = zap.NewNop()
+	}
+	cfg := &routingPolicyFilter{
+		logger:                 logger,
+		isDefault:              p.IsDefault,
+		allowedTrafficTypes:    p.AllowedTrafficTypes,
+		allowedTrafficTypeMask: 0,
+	}
+	p.RoutingPolicyHandler.Subscribe(cfg.onRoutingPolicyConfigUpdate)
+	return producer.NewFilterFunc(cfg.Filter, producer.RoutingPolicyFilter, configSource)
+}
+
+type routingPolicyFilter struct {
+	logger                 *zap.Logger
+	isDefault              bool
+	allowedTrafficTypes    []string
+	allowedTrafficTypeMask uint64
+}
+
+func (f *routingPolicyFilter) Filter(m producer.Message) bool {
+	msg, ok := m.(message)
+	if !ok || msg.rp.TrafficTypes == 0 {
+		return f.isDefault
+	}
+	mask := atomic.LoadUint64(&f.allowedTrafficTypeMask)
+	return msg.rp.TrafficTypes&mask != 0
+}
+
+func (f *routingPolicyFilter) onRoutingPolicyConfigUpdate(policyConfig routing.PolicyConfig) {
+	trafficTypes := policyConfig.TrafficTypes()
+	mask := uint64(0)
+	for _, trafficType := range f.allowedTrafficTypes {
+		bitPosition := f.resolveTrafficTypeToBitPosition(trafficTypes, trafficType)
+		if bitPosition == -1 {
+			f.logger.Warn("traffic type not found in routing policy config", zap.String("missing_traffic_type", trafficType))
```

Comment:
Emit a metric here as well ? This would help with better tracking or alerting of the issue.